File: BackEnd/src/main/java/com/Checkerly/BackEnd/services/OrganizerService.java
package com.Checkerly.BackEnd.services;

import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.Checkerly.BackEnd.domain.Organizer;
import com.Checkerly.BackEnd.dto.OrganizerDTO;
import com.Checkerly.BackEnd.repository.OrganizerRepository;
import com.Checkerly.BackEnd.services.exception.ObjectNotFoundException;

@Service
public class OrganizerService {

	@Autowired
    private OrganizerRepository repo;
	
	public List<Organizer> findAll(){
		return repo.findAll();
	}
	
	public Organizer findById(String id) {
	    Optional<Organizer> Organizer = repo.findById(id);
	    return Organizer.orElseThrow(() -> new ObjectNotFoundException("Objeto não encontrado"));
	}
	
	public Organizer insert(Organizer obj) {
		Organizer savedOrganizer = repo.save(obj);
		System.out.println("Organizer inserted: " + savedOrganizer);
		return savedOrganizer;
	}
	
	public void delete(String id) {
		findById(id);
		repo.deleteById(id);
	}
	
	public Organizer update(Organizer obj) {
		Organizer newObj = findById(obj.getId());
		updateData(newObj, obj);
		return repo.save(newObj);
	}
	
	private void updateData(Organizer newObj, Organizer obj) {
		newObj.setName(obj.getName());
		newObj.setEmail(obj.getEmail());
	}

	public Organizer fromDTO(OrganizerDTO objDto) {
		return new Organizer(objDto.getId(), objDto.getName(),objDto.getSenha(), objDto.getEmail(), objDto.getCelular());
	}
}

File: BackEnd/src/main/kotlin/com/Checkerly/BackEnd/services/OrganizerService.kt
package com.Checkerly.BackEnd.services

import com.Checkerly.BackEnd.domain.Organizer
import com.Checkerly.BackEnd.dto.OrganizerDTO
import com.Checkerly.BackEnd.repository.OrganizerRepository
import com.Checkerly.BackEnd.services.exception.ObjectNotFoundException
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service
import java.util.Optional

@Service
class OrganizerService {
    @Autowired
    private val repo: OrganizerRepository? = null

    @Autowired
    private val passwordEncoder: PasswordEncoder? = null

    fun findAll(): List<Organizer?>? {
        return repo.findAll()
    }

    fun findById(id: String?): Organizer? {
        val Organizer: Optional<Organizer?>? = repo.findById(id)
        return Organizer.orElseThrow({ ObjectNotFoundException("Objeto não encontrado") })
    }

    fun insert(obj: Organizer): Organizer? {
        val encodedPassword: String? = passwordEncoder.encode(obj.getSenha())
        obj.setSenha(encodedPassword)
        return repo.save(obj)
    }

    fun delete(id: String?) {
        findById(id)
        repo.deleteById(id)
    }

    fun update(obj: Organizer): Organizer? {
        val newObj: Organizer? = findById(obj.getId())
        updateData(newObj, obj)
        return repo.save(newObj)
    }

    private fun updateData(newObj: Organizer, obj: Organizer) {
        newObj.setName(obj.getName())
        newObj.setEmail(obj.getEmail())
    }

    fun fromDTO(objDto: OrganizerDTO): Organizer? {
        return Organizer(objDto.getId(), objDto.getName(), objDto.getSenha(), objDto.getEmail(), objDto.getCelular())
    }
}
